exercises/practice/rest-api/rest-api.spec.js

```diff
@@ -0,0 +1,345 @@
+import { RestAPI } from './rest-api';
+
+describe('Rest API', () => {
+  describe('user management', () => {
+    test('no users', () => {
+      const restAPI = new RestAPI({ users: [] });
+      expect(restAPI.get('/users')).toEqual({ users: [] });
+    });
+    test('add user', () => {
+      const restAPI = new RestAPI({ users: [] });
+      expect(restAPI.post('/add', { user: 'Adam' })).toEqual({
+        name: 'Adam',
+        owes: {},
+        owed_by: {},
+        balance: 0.0,
+      });
+    });
+    test('get single user', () => {
+      const restAPI = new RestAPI({
+        users: [
+          {
+            name: 'Adam',
+            owes: {},
+            owed_by: {},
+            balance: 0.0,
+          },
+          {
+            name: 'Bob',
+            owes: {},
+            owed_by: {},
+            balance: 0.0,
+          },
+        ],
+      });
+      expect(restAPI.get('/users?users=Bob')).toEqual({
+        users: [
+          {
+            name: 'Bob',
+            owes: {},
+            owed_by: {},
+            balance: 0.0,
+          },
+        ],
+      });
```

Comment:
General comment regarding test file: I would recommend thinking about the readability of the tests. The statements currently span so many lines that is is hard to grasp the actual statements. I think it would help to maybe write some of the objects in one line and to use variables to define them. Here an example how this could look like applying both ideas:
```javascript
      const seedUsers = [
        { name: 'Adam', owes: {}, owed_by: {}, balance: 0.0 },
        { name: 'Bob', owes: {}, owed_by: {}, balance: 0.0 },
      ];
      const restAPI = new RestAPI({ users: seedUsers });
```